src/renderer/components/SettingsModal.tsx

```diff
@@ -322,15 +324,15 @@ const PythonSettings = memo(() => {
     );
 
     const [ncnnGPU, setNcnnGPU] = useNcnnGPU;
-    const [fullGpuList, setFullGpuList] = useState<string[]>([]);
+    const [ncnnGpuList, setNcnnGpuList] = useState<string[]>([]);
     useAsyncEffect(
         {
             supplier: async () => {
-                const fullGpuInfo = await ipcRenderer.invoke('get-gpu-info');
-                const gpuNames = fullGpuInfo.controllers.map((g) => g.model);
+                const ncnnGpuInfo = await backend.listNcnnGpus();
+                const gpuNames = Object.values(ncnnGpuInfo);
```

Comment:
Why did you make the backend return a `Record<number, string>` when you assume it's indexed like a `string[]` here?

However, I'm not sure which of the two is the problem. Is the backend more complex than it needs to be, or is the frontend making an assumption it shouldn't make?

Reply:
I could have returned a list from the backend, but I wanted to ensure order. I can change this to whatever you think is the proper way

Reply:
> but I wanted to ensure order

Aren't lists ordered? Or asked in another way: What order did you want to ensure?

Reply:
I just wanted to make sure no f*ckery happens between the JSON conversion and parsing. If that's guaranteed to be correct then I'll just use a list. I'm trying to preserve gpu index order

Reply:
😂 array it is, then